server/src/test/java/org/elasticsearch/index/fielddata/AbstractFieldDataTestCase.java

```diff
@@ -90,7 +90,7 @@ public <IFD extends IndexFieldData<?>> IFD getForField(String type, String field
             if (docValues) {
                 fieldType = new KeywordFieldMapper.Builder(fieldName, IndexVersion.current()).build(context).fieldType();
             } else {
-                fieldType = new TextFieldMapper.Builder(fieldName, createDefaultIndexAnalyzers()).fielddata(true)
+                fieldType = new TextFieldMapper.Builder(fieldName, createDefaultIndexAnalyzers(), false).fielddata(true)
```

Comment:
I think `indexService.getIndexSettings().getMode().isSyntheticSourceEnabled()` can be used here instead?